akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java

```diff
@@ -627,7 +627,8 @@ public void failOnIncorrectExpectedStateForThenRun() {
 
     new EventFilter(Logging.Error.class, Adapter.toUntyped(testKit.system()))
         .occurrences(1)
-        .message("java.lang.Integer cannot be cast to java.lang.String")
+        // the error messages slightly changed in later JDKs
+        .matches("(class )?java.lang.Integer cannot be cast to (class )?java.lang.String.*")
```

Comment:
Will it fail if not against 11?

Reply:
No (it succeeded the PR validation, for example).

The difference is that JDK 11 is more polite about calling those things by including their title: ~Mister~ `class java.lang.Integer` and ~Miss~ `class java.lang.String`. The change above uses a regex to make matching that extra `class` optional.